java/src/main/java/ai/rapids/cudf/RegexFlags.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *  Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+/**
+ * Regex flags setting, closely following cudf::strings::regex_flags.
+ * These types can be or'd to combine them. The values are chosen to
+ * leave room for future flags and to match the Python flag values.
+ */
+public enum RegexFlags {
```

Comment:
cudf::strings::regex_flags does not map well to an enum in java. The flags are actually bit sets that can be combined to produce a single result, i.e.

`regex_flags.MULTILINE | regexp_flags.ASCII` would make for a regular expression that only operated on ASCII but did multiline.

For us I think we want to rename this to be singular `RegexFlag` and then instead of passing in a `RegexFlag` tot he constructor of `RegexProgram` we pass in an `EnumSet<RegexFlag>`. This would provide the same semantics of the C++ code `EnumSet.of(RegexFlag.MULTILINE, RegexFlag.ASCII)`.  When we pass the results down to C++ we would want to combine them with a but OR and just pass down a single int.

```
int allFlags = 0;
flags.forEach(v => allFlags | v.nativeId);
```